doc/specs/stdlib_stats_distribution.md

```diff
@@ -0,0 +1,456 @@
+---
+title: stats_distribution
+---
+
+# Statistical Distribution
+
+[TOC]
+
+## `random_seed` - set or get a value of seed to the random distribution psudorandom number generator
+
+### Status
+
+Experimental
+
+### Description
+
+Must set the seed value before calling random distribution for variates. 
+
+### Syntax
+
+`call random_seed(put [, get])`
+
+### Arguments
+
+`put`: argument has intent `in` and may be a scalar of type `integer` with kind of int32.
+
+`get`: optional argument has intent `out` and is a scalar of type `integer` with kind of int32.
+
+### Return value
+
+
+### Example
+
+```fortran
+program demo_norm_seed
+    use stdlib_stats_distribution, only : random_seed
+    use iso_fortran_env, only : int32
+    implicit none
+    integer(int32) :: seed, seed_value
+
+    seed = 1234567
+    call random_seed(seed)                   ! reset the value of seed
+    call random_seed(seed, seed_value)       ! get current value of seed
+end program demo_norm_seed
+```
+
+## `uniform_distribution_rvs` - uniform distribution random variates
+
+### Status
+
+Experimental
+
+### Description
```

Comment:
Would it be possible to return an muti-dimension array (e.g., a ran-3 array), e.g., as in [Matlab](https://www.mathworks.com/help/stats/unifrnd.html)

Reply:
> Would it be possible to return an muti-dimension array (e.g., a ran-3 array), e.g., as in [Matlab](https://www.mathworks.com/help/stats/unifrnd.html)

Yes. the functions are elemental except the the random number generators, they should work on multi-dimension arrays.

Reply:
It would be nice that the random number generator functions ("_rvs") also support arrays up to rank 7 or 15 (depending on the compiler). This could be easily generated with Fypp (maybe in a next PR?).

Reply:
random number generator function itself has no issue with array support, it is elemental. The problem is how to initiate the function call. Normally, one uses one set of parameters to get either one or an array of random numbers. This is how current implementation worked. If we want array support up to rank 15, then we need a mask array as an optional argument to the function call. If peoples like the idea, we can certainly implement it next round.

Reply:
I was thinking to extend the rank-1 function to multiple ranks, e.g. with multiple loops. Therefore, there is no need for optional arguments (there would rank+1 different procedures). But, as you mentioned, we can think about that in a next round if desired.